AppKit/CoreAnimation/CPAnimationContext.j

```diff
@@ -677,31 +677,44 @@ FrameUpdater.prototype.addTarget = function(target, keyPath, duration)
 
 var createUpdateFrame = function(aView, aKeyPath)
 {
-    if (aKeyPath !== "frame" && aKeyPath !== "frameSize")
+    if (aKeyPath !== "frame" && aKeyPath !== "frameSize" && aKeyPath !== "frameOrigin")
         return nil;
 
     var style = getComputedStyle(aView._DOMElement),
           getCSSPropertyValue = function(prop) {
                 return ROUND(parseFloat(style.getPropertyValue(prop)));
-          };
+          },
 
-    var updateFrame = function(timestamp)
+        initialFrame  = [aView frame],
+        initialX      = initialFrame.origin.x,
+        initialY      = initialFrame.origin.y,
+        initialWidth  = initialFrame.size.width,
+        initialHeight = initialFrame.size.height,
```

Comment:
Just keep the initialFrame. Make a copy of it, so it will not be updated with the animation